src/System.Runtime/tests/System/Enum.cs

```diff
@@ -4,219 +4,165 @@
+    [Theory]
+    [InlineData("Red", false, true, SimpleEnum.Red)]
+    [InlineData(" Red", false, true, SimpleEnum.Red)]
+    [InlineData(" red ", true, true, SimpleEnum.Red)]
+    [InlineData(" Red , Blue ", false, true, (SimpleEnum)3)]
 
-            b = Enum.TryParse<SimpleEnum>("Purple", out e);
-            Assert.True(!b);
+    [InlineData("1", false, true, SimpleEnum.Red)]
+    [InlineData(" 1 ", false, true, SimpleEnum.Red)]
+    [InlineData("2", false, true, SimpleEnum.Blue)]
+    [InlineData("99", false, true, (SimpleEnum)99)]
 
-            b = Enum.TryParse<SimpleEnum>("1", out e);
-            Assert.True(b);
-            Assert.Equal(e, SimpleEnum.Red);
-
-            b = Enum.TryParse<SimpleEnum>(" 1 ", out e);
-            Assert.True(b);
-            Assert.Equal(e, SimpleEnum.Red);
-
-            b = Enum.TryParse<SimpleEnum>("2", out e);
-            Assert.True(b);
-            Assert.Equal(e, SimpleEnum.Blue);
-
-            b = Enum.TryParse<SimpleEnum>("99", out e);
-            Assert.True(b);
-            Assert.Equal(e, (SimpleEnum)99);
-
-            return;
-        }
-    }
-
-    [Fact]
-    public static void TestGetName()
+    [InlineData(" red ", false, false, (SimpleEnum)999)] // No actual expected result
+    [InlineData("Purple", false, false, (SimpleEnum)999)]
+    public static void TestParse(string value, bool ignoreCase, bool expectedCanParse, SimpleEnum expectedParseResult)
     {
-        String s;
+        bool b;
+        SimpleEnum e;
+        if (!ignoreCase)
         {
-            Type t = typeof(SimpleEnum);
-            s = Enum.GetName(t, 99);
-            Assert.Equal(s, null);
-
-            s = Enum.GetName(t, 1);
-            Assert.Equal(s, "Red");
-
-            s = Enum.GetName(t, SimpleEnum.Red);
-            Assert.Equal(s, "Red");
-
-            // In the case of multiple matches, GetName returns one of them (which one is an implementation detail.)
-            s = Enum.GetName(t, 3);
-            Assert.True(s == "Green" || s == "Green_a" || s == "Green_b");
+            b = Enum.TryParse<SimpleEnum>(value, out e);
+            Assert.Equal(expectedCanParse, b);
+            if (expectedCanParse)
+            {
+                Assert.Equal(expectedParseResult, e);
+            }
         }
 
-        // Negative tests
+        b = Enum.TryParse<SimpleEnum>(value, ignoreCase, out e);
+        Assert.Equal(expectedCanParse, b);
+        if (expectedCanParse)
         {
-            Type t = typeof(SimpleEnum);
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(null, 1));
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(t, null));
-            Assert.Throws<ArgumentNullException>(() => Enum.GetName(typeof(Object), null));
-            Assert.Throws<ArgumentException>(() => Enum.GetName(t, "Red"));
-            Assert.Throws<ArgumentException>(() => Enum.GetName(t, (IntPtr)0));
+            Assert.Equal(expectedParseResult, e);
```

Comment:
Ditto, this can be:
```C
Assert.Equal(expectedCanParse, Enum.TryParse<SimpleEnum>(value, ignoreCase, out e));
Assert.Equal(expectedParseResult, e);
```